mne_bids/utils.py

```diff
@@ -363,6 +364,51 @@ def _age_on_date(bday, exp_date):
     return exp_date.year - bday.year - 1
 
 
+def _bday_on_age(age, exp_date):
+    """Calculate birthday from age and experiment date.
```

Comment:
can you clarify what you do here? I am lost.

Reply:
An error was there. I allowed exp_date to be None, but looking again at `write.py`, I see that's not how it's written in. 

This function is meant to calculate the birthday tuple from age/exp_date. If age is a fraction then there's some additional logic to compute the month/day, else one just defaults to (1, 1).

Reply:
If this is okay, then this can get reviewed again.

Reply:
where is this code tested? I find it dangerous to guess birthday from age assuming it can be a float. Have you seen this somewhere? Also as you return a date I would output a datetime instance.

Reply:
This code is tested in `test_read.py`

```
    if check_version('mne', '0.20'):
        # if measurement date is float
        raw = mne.io.read_raw_fif(raw_fname, verbose=False)
        raw = raw.set_meas_date(datetime(2000, 1, 1, tzinfo=timezone.utc))
        write_raw_bids(raw, bids_basename, bids_root, overwrite=True,
                       verbose=False)
        participants_tsv_fpath = op.join(bids_root, 'participants.tsv')
        participants_tsv = _from_tsv(participants_tsv_fpath)
        participants_tsv['age'][0] = 5.5
        _to_tsv(participants_tsv, participants_tsv_fpath)
        raw = read_raw_bids(bids_fname, Path(bids_root))
        print(raw.info['subject_info'])
        assert raw.info['subject_info']['birthday'][0] == 1994
        assert raw.info['subject_info']['birthday'][1] == 6
```

I can remove it if you think it's unnecessary to go beyond re-computing the year. 

Regarding date: https://mne.tools/stable/generated/mne.Info.html seems to require birthdate to be a tuple of ints though?

Reply:
so in bids .tsv file you expect some people do have an age as a float? to me the intention was to have the age as an int so I would enforce that birthday in Jan 1st so that IO round trip is correct.

then your function is just

(int(exp_date.year - age), 1, 1)